prometheus_speedtest/prometheus_speedtest.py

```diff
@@ -131,7 +140,8 @@ def main(argv):
         return
 
     registry = core.CollectorRegistry(auto_describe=False)
-    registry.register(SpeedtestCollector(servers=FLAGS.servers))
+    registry.register(
```

Comment:
Can you add a check to ensure that includes is not a subset of excludes? Something like this.

if not set(servers) - set(excludes):
    logging.fatal('--servers is a subset of --excludes, does not match anything.')

Reply:
Actually this is better:

```python
if set(servers).issubset(excludes):
    logging.fatal('--servers is a subset of --excludes, does not match anything.')
```

Reply:
I will double check, but while I was researching the excludes flag, I believe the speed test project does this check. If not I will add your suggestion.

Reply:
It does do a check when you go to run a Speedtest, but it doesn't crash prometheus_speedtest when it fails. I'd rather it crashed right away so that you know something is wrong.

```python
$ python3 -m prometheus_speedtest.prometheus_speedtest --servers 1771 --excludes 1771
I1029 23:11:39.101155 4435500480 prometheus_speedtest.py:150] Starting HTTP server listening on 0.0.0.0:9516
I1029 23:11:43.850766 123145484587008 prometheus_speedtest.py:127] GET /probe HTTP/1.1 - ('127.0.0.1', 54776)
I1029 23:11:43.850970 123145484587008 prometheus_speedtest.py:54] Performing Speedtest...
----------------------------------------
Exception happened during processing of request from ('127.0.0.1', 54776)
Traceback (most recent call last):
  File "/usr/local/Cellar/python@3.8/3.8.6/Frameworks/Python.framework/Versions/3.8/lib/python3.8/socketserver.py", line 650, in process_request_thread
    self.finish_request(request, client_address)
  File "/usr/local/Cellar/python@3.8/3.8.6/Frameworks/Python.framework/Versions/3.8/lib/python3.8/socketserver.py", line 360, in finish_request
    self.RequestHandlerClass(request, client_address, self)
  File "/Users/jraviles/Projects/src/github.com/jeanralphaviles/prometheus_speedtest/prometheus_speedtest/prometheus_speedtest.py", line 118, in __init__
    super().__init__(directory=static_directory, *args, **kwargs)
  File "/usr/local/Cellar/python@3.8/3.8.6/Frameworks/Python.framework/Versions/3.8/lib/python3.8/http/server.py", line 647, in __init__
    super().__init__(*args, **kwargs)
  File "/usr/local/Cellar/python@3.8/3.8.6/Frameworks/Python.framework/Versions/3.8/lib/python3.8/socketserver.py", line 720, in __init__
    self.handle()
  File "/usr/local/Cellar/python@3.8/3.8.6/Frameworks/Python.framework/Versions/3.8/lib/python3.8/http/server.py", line 427, in handle
    self.handle_one_request()
  File "/usr/local/Cellar/python@3.8/3.8.6/Frameworks/Python.framework/Versions/3.8/lib/python3.8/http/server.py", line 415, in handle_one_request
    method()
  File "/Users/jraviles/Projects/src/github.com/jeanralphaviles/prometheus_speedtest/prometheus_speedtest/prometheus_speedtest.py", line 130, in do_GET
    prometheus_client.MetricsHandler.do_GET(self)
  File "/usr/local/lib/python3.8/site-packages/prometheus_client/exposition.py", line 169, in do_GET
    status, header, output = _bake_output(registry, accept_header, params)
  File "/usr/local/lib/python3.8/site-packages/prometheus_client/exposition.py", line 40, in _bake_output
    output = encoder(registry)
  File "/usr/local/lib/python3.8/site-packages/prometheus_client/exposition.py", line 106, in generate_latest
    for metric in registry.collect():
  File "/usr/local/lib/python3.8/site-packages/prometheus_client/registry.py", line 82, in collect
    for metric in collector.collect():
  File "/Users/jraviles/Projects/src/github.com/jeanralphaviles/prometheus_speedtest/prometheus_speedtest/prometheus_speedtest.py", line 85, in collect
    results = self._tester.test()
  File "/Users/jraviles/Projects/src/github.com/jeanralphaviles/prometheus_speedtest/prometheus_speedtest/prometheus_speedtest.py", line 59, in test
    client.get_servers(servers=self._servers, exclude=self._excludes))
  File "/usr/local/lib/python3.8/site-packages/speedtest.py", line 1349, in get_servers
    raise NoMatchedServers()
speedtest.NoMatchedServers
```

Reply:
OK thanks, I will make your suggestion

Reply:
updated